src/derivative_operators/convolutions.jl

```diff
@@ -24,65 +25,66 @@ end
 
 function convolve_BC_left!(x_temp::AbstractVector{T}, x::AbstractVector{T}, A::DerivativeOperator{T,S}) where {T<:Real,S<:SVector}
     coeffs = A.low_boundary_coefs
-    Threads.@threads for i in 1 : A.boundary_length
+    Threads.@threads for i in 1 : A.boundary_point_count
         xtempi = coeffs[i][1]*x[1]
-        @inbounds for idx in 2:A.stencil_length
+        @inbounds for idx in 2:A.boundary_stencil_length
             xtempi += coeffs[i][idx] * x[idx]
         end
-        x_temp[i] = xtempi
+        x_temp[i+1] = xtempi
     end
 end
 
 function convolve_BC_right!(x_temp::AbstractVector{T}, x::AbstractVector{T}, A::DerivativeOperator{T,S}) where {T<:Real,S<:SVector}
     coeffs = A.high_boundary_coefs
-    Threads.@threads for i in 1 : A.boundary_length
+    Threads.@threads for i in 1 : A.boundary_point_count
         xtempi = coeffs[i][end]*x[end]
-        @inbounds for idx in A.stencil_length-1:-1:1
+        @inbounds for idx in A.boundary_stencil_length:-1:1
             xtempi += coeffs[i][end-idx] * x[end-idx]
         end
-        x_temp[end-A.boundary_length+i] = xtempi
+        x_temp[end-A.boundary_point_count+i-1] = xtempi
     end
 end
 
 ###########################################
+# NOT NEEDED ANYMORE
```

Comment:
These are definitely needed, otherwise the operator will be slower due to the indexing fallback.

Reply:
ok.